domain-services/ifix-master-data-service/src/main/java/org/egov/migration/V20210915104900_MongoIndexScript.java

```diff
@@ -0,0 +1,87 @@
+@Slf4j
+public class V20210915104900_MongoIndexScript {
+
+    public static final String MDS_INDEX_AUTHOR = "iFIX-Master-Data-Service";
+
+    /**
+     * @param db
+     */
+    @ChangeSet(id = "V20210914162000_createGovernmentIndex", author = MDS_INDEX_AUTHOR, order = "001")
+    public void createGovernmentIndex(MongoDatabase db) {
+        MongoCollection<Document> mongoCollection = db.getCollection("government");
+
+        mongoCollection.createIndex(Indexes.ascending("name"));
+    }
+
+    /**
+     * @param db
+     */
+    @ChangeSet(id = "V20210914164000_createDepartmentIndex", author = MDS_INDEX_AUTHOR, order = "002")
+    public void createDepartmentIndex(MongoDatabase db) {
+        MongoCollection<Document> mongoCollection = db.getCollection("department");
+
+        mongoCollection.createIndex(Indexes.ascending("name"));
+        mongoCollection.createIndex(Indexes.ascending("code"));
+        mongoCollection.createIndex(Indexes.ascending("tenantId"));
+    }
+
+    /**
+     * @param db
+     */
+    @ChangeSet(id = "V20210914174400_createChartOfAccountIndex", author = MDS_INDEX_AUTHOR, order = "003")
+    public void createChartOfAccountIndex(MongoDatabase db) {
+        MongoCollection<Document> mongoCollection = db.getCollection("chartOfAccount");
+
+        mongoCollection.createIndex(Indexes.ascending("coaCode"));
+        mongoCollection.createIndex(Indexes.ascending("majorHead"));
+        mongoCollection.createIndex(Indexes.ascending("tenantId"));
+        mongoCollection.createIndex(Indexes.ascending("subMajorHead"));
+        mongoCollection.createIndex(Indexes.ascending("minorHead"));
+        mongoCollection.createIndex(Indexes.ascending("subHead"));
+        mongoCollection.createIndex(Indexes.ascending("groupHead"));
+        mongoCollection.createIndex(Indexes.ascending("objectHead"));
+    }
+
+    /**
+     * @param db
+     */
+    @ChangeSet(id = "V20210914175000_createExpenditureIndex", author = MDS_INDEX_AUTHOR, order = "004")
+    public void createExpenditureIndex(MongoDatabase db) {
+        MongoCollection<Document> mongoCollection = db.getCollection("expenditure");
+
+        mongoCollection.createIndex(Indexes.ascending("name"));
+        mongoCollection.createIndex(Indexes.ascending("code"));
+        mongoCollection.createIndex(Indexes.ascending("tenantId"));
+    }
+
+    /**
+     * @param db
+     */
+    @ChangeSet(id = "V20210914175400_createProjectIndex", author = MDS_INDEX_AUTHOR, order = "005")
+    public void createProjectIndex(MongoDatabase db) {
+        MongoCollection<Document> mongoCollection = db.getCollection("project");
+
+        mongoCollection.createIndex(Indexes.ascending("name"));
+        mongoCollection.createIndex(Indexes.ascending("code"));
+        mongoCollection.createIndex(Indexes.ascending("tenantId"));
+        mongoCollection.createIndex(Indexes.ascending("expenditureId"));
+        mongoCollection.createIndex(Indexes.ascending("departmentId"));
```

Comment:
Add index for ancestry departmentEntityId.

Reply:
done